streams/src/main/java/org/apache/kafka/streams/state/internals/CachingSessionStore.java

```diff
@@ -50,8 +51,10 @@
     private long maxObservedTimestamp; // Refers to the window end time (determines segmentId)
 
     CachingSessionStore(final SessionStore<Bytes, byte[]> bytesStore,
-                        final long segmentInterval) {
+                        final long segmentInterval,
+                        final boolean dslUsage) {
         super(bytesStore);
+        this.dslUsage = dslUsage;
```

Comment:
Potentially better to put this after L59 because dslUsage is ranked later of `cacheFuncion` in parameter list, and later of `segmentInterval` in argument list.